internal/initializer/crds.go

```diff
@@ -131,6 +134,45 @@ func (c *CoreCRDs) Run(ctx context.Context, kube client.Client) error {
 		if err := pa.Apply(ctx, crd); err != nil {
 			return errors.Wrap(err, "cannot apply crd")
 		}
+
+		crdNames = append(crdNames, crd.Name)
+	}
+
+	for _, crdName := range crdNames {
+		if err := waitForCRDEstablished(ctx, kube, crdName); err != nil {
+			return errors.Wrapf(err, "CRD %s not established", crdName)
+		}
 	}
+
 	return nil
 }
+
+func waitForCRDEstablished(ctx context.Context, kube client.Client, crdName string) error {
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+	timeout := time.After(60 * time.Second)
+
+	for {
```

Comment:
Better use client-go's `wait.PollUntilContextTimeout`.

Reply:
@sttts Seems like this was present in api machinery. I've added this. Can you please check?